packages/side-nav/src/vaadin-side-nav.js

```diff
@@ -105,6 +105,31 @@ class SideNav extends FocusMixin(ElementMixin(ThemableMixin(PolylitMixin(LitElem
         notify: true,
         reflectToAttribute: true,
       },
+
+      /**
+       * The object used to localize child items.
+       *
+       * To change the default localization, replace the entire
+       * `i18n` object with a custom one.
+       *
+       * The object has the following structure and default values:
+       * ```
+       * {
+       *   toggle: 'Toggle child items'
+       * }
+       * ```
+       *
+       * @type {SideNavItemI18n}
+       * @default {English/US}
+       */
+      i18n: {
```

Comment:
Currently side nav items only propagate their i18n to their children, be we also want to propagate from side nav to its items. We should also add a test for that scenario.

Apart from that setting an i18n on a parent item overrides the i18n set on a child item. I wonder if for a first step we should keep it simple and only allow setting the i18n on the side nav itself, rather than having (public) API that allows you to set the i18n on each level? Basically just make the i18n property on side nav items non-public.

Reply:
> setting an i18n on a parent item overrides the i18n set on a child item. I wonder if for a first step we should keep it simple and only allow setting the i18n on the side nav itself

We do have `vaadin-avatar` and `vaadin-avatar-group` that both expose `i18n` publicly, and there it's also used for screen readers. Also, having both components with public property makes it easy to put it to a shared mixin.